### DISCUSSION TITLE
Using Evidently in Airflow using Docker

### DISCUSSION THREAD

**Discussion by PranavKhedkar** [apache/airflow#39024] [Q&A] [answered]
I am using airflow in a docker container. I installed evidently in airflow but when I try to use it, there is an error stating that "from pydantic.v1.fields import SHAPE_DICT
ModuleNotFoundError: No module named 'pydantic.v1.fields'; 'pydantic.v1' is not a package". Is this due to a version conflict? I tried upgrading as well as downgrading pydantic, but I face the same error

**Comment by phi-friday** [comment]
It would be helpful if you could let us know which versions of `airflow` and `pydantic` you are using.

**Reply by PranavKhedkar** [reply]
Here are the versions:
airflow : 2.9.0
pydantic : 1.10.15
evidently : 0.4.7

**Reply by phi-friday** [reply]
`airflow` defines the modules associated with `pydantic` in [here](https://github.com/apache/airflow/blob/v2-9-stable/airflow/utils/pydantic.py).

As you'll see if you check the file, it can work regardless of the version of `pydantic`.
I installed `pydantic@v1` as you suggested, and it works fine.
(I didn't really run it, I just installed `pydantic@v1` and tried out the module.)

Based on the error message you gave first, it seems that certain libraries only support `pydantic@v2`.
If you could tell us what other libraries you have installed, we might be able to help you further.

Separately, `airflow` recommends `pydantic@v2`
> [link](https://github.com/apache/airflow/blob/50f22ffe9150e93f7fc08a9dc8a48880daad374b/hatch_build.py#L118). 
```yaml
    "pydantic": [
        "pydantic>=2.3.0",
    ],
```

If you want to install `airflow` and `pydantic` together, install `apache-airflow[pydantic]`.

**Reply by Taragolis** [reply]
> airflow defines the modules associated with pydantic in [here](https://github.com/apache/airflow/blob/v2-9-stable/airflow/utils/pydantic.py).

This ones for internal usage only

**Comment by PranavKhedkar** [comment]
The libraries that I am using:
pandas
sklearn
datetime
evidently

**Comment by PranavKhedkar** [comment, marked as answer]
I got the solution by using these versions:
airflow == 1.10.9
evidently == 0.2.0
